pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/DirectMemoryManager.java

```diff
@@ -26,8 +29,13 @@
   /**
    * @see RealtimeIndexOffHeapMemoryManager
    */
+  public DirectMemoryManager(final String segmentName, ServerMetrics serverMetrics) {
+    super(serverMetrics, segmentName);
+  }
+
+  @VisibleForTesting
   public DirectMemoryManager(final String segmentName) {
-    super(segmentName);
+    this(segmentName, new ServerMetrics(new MetricsRegistry()));
```

Comment:
The server metrics could be null for clients that don't care about it, but that means you will have null check against it. Up to you.